plugins/inputs/snmp_trap/snmp_trap.go

```diff
@@ -104,6 +103,9 @@ func (s *SnmpTrap) Init() error {
 			return err
 		}
 	case "netsnmp":
+		if s.Timeout == 0 {
+			s.Timeout = defaultTimeout
+		}
```

Comment:
@DStrand1 I think you need to remove this section as the default should be already set in `init`, isn't it?

Reply:
I thought it would be better to leave it this way in case we decide later to deprecate this flag

Reply:
No, please remove this as it now does not allow the user to enforce an infinite timeout.